src/main.js

```diff
@@ -1,14 +1,28 @@
 const Apify = require('apify');
-const { handlGithubIssues } = require('./routes');
-const { getGithubIssuesUrls } = require('./tools');
+const { handleGithubIssues } = require('./issues-scraper');
+const { sendModifiedIssuesNotification } = require('./slack-notifier');
+const { getGithubIssuesRequests, getModifiedIssues } = require('./tools');
+const { ISSUES_STATE, ISSUES_KEY_VALUE_STORE } = require('./constants');
 
 const { utils: { log } } = Apify;
 
 Apify.main(async () => {
-    const { repositories, token, channel, proxyConfiguration } = await Apify.getInput();
+    const input = await Apify.getInput();
+    const { repositories, token, separateNotification, proxyConfiguration, excludeOpenedIssues, excludeClosedIssues } = input;
 
-    const issuesUrls = getGithubIssuesUrls(repositories);
-    const requestList = await Apify.openRequestList('github-issues-urls', issuesUrls);
+    // handle missing '#' at the beginning of channel name
+    const channel = input.channel[0] === '#' ? input.channel : `#${input.channel}`;
+
+    const issuesStore = await Apify.openKeyValueStore(ISSUES_KEY_VALUE_STORE);
+
+    const issuesState = await issuesStore.getValue(ISSUES_STATE) || {};
+    Apify.events.on('persistState', async () => { await issuesStore.setValue(ISSUES_STATE, issuesState); });
```

Comment:
I think the new implementation is wrong, you loading from and updating to `ISSUES_KEY_VALUE_STORE` & `ISSUES_STATE` and then you compare with the same record. That means you will compare two indentical records. Let me know if I overlooked it.

What I meant to propose is to keep both the local unnamed and named storage. The local is used to keep the state of this run (like you had it before) and this state is then compared with the named KV record. Once you are done comparing, you overwrite the named KV record with the local state.

Reply:
Oh, I missed the point that we want to keep both default and named storages. And you're right, I'm comparing the same records, I didn't replace the local KVS with the named KVS properly. I tested it by running the actor and it worked as I expected so maybe it worked by accident because of the missing `await` before `setValue` function.

I'll change the implementation to use both local and named KVS as you described 🙂